swap_meet/vendor.py

```diff
@@ -1,2 +1,111 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        """Removes item from this vendor's inventory, adds it to the other vendor's inventory and vice versa."""
+
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            item_to_friend = self.remove(my_item)
+            other_vendor.add(item_to_friend)
+            item_to_me = other_vendor.remove(their_item)
+            self.add(item_to_me)
+            return True
+        # If item isn't in the inventory
+        return False
```

Comment:
[suggestion] I like inverting the logic of the condition to be able to treat it like a guard clause, which allows us to outdent the main logic of the function something like the following:

```python
        if my_item not in self.inventory or others_item not in other_vendor.inventory:
            return False

        # Everything else is now less indented
```

Note that this uses [De Morgan's Law](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). Basically, `x and y` is equal to `not x or not y`